docs/paper/dev/api/how-do-plugins-work.md

```diff
@@ -0,0 +1,101 @@
+
+Some events are cancellable. This means that when the event is fired, it can be cancelled which negates or stops the 
+effect of the event. For example, the `PlayerMoveEvent` is cancellable. This means that when it is cancelled, the player 
+will not move. This is useful for things like anti-cheat, where you want to cancel the event if the player is moving too fast.
+
+It is important to think about how "hot" an event is when writing event listeners. A "hot" event is an event that is fired
+very often. For example, the `PlayerMoveEvent` is fired every time a player moves. This means that if you have a lot of
+expensive code in your event listener, it will be run every time a player moves. This can cause a lot of lag. It is
+important to keep event listeners as lightweight as possible.
+
+## Commands
+
+Commands are a way for players to run code on the server. Commands are registered by plugins and can be run by players.
+For example, the `/help` command is registered by the server and can be run by players. Commands can be run by players
+by typing them in the chat or by running them from a command block. Commands can also be run by other plugins.
+
+Commands can have arguments. For example, the `/give` command takes an argument for the player to give the item to and
+an argument for the item to give. Arguments are separated by spaces. For example, the command `/give Notch diamond` will
+give the player named Notch a diamond. Note here that the arguments are `["Notch", "diamond"]`.
+
+### Permissions
+
+Permissions are a way to control who can run commands and who can listen to events. Permissions 
+are registered by plugins and can be checked by other plugins. Permissions can be granted to players and groups. 
+Permissions have a hierarchical nature. For example, the `paper.command.help` permission is a sub-permission to 
+`paper.command`. This means that if a player has the `paper.command` permission, they will also have the 
+`paper.command.help` permission.
+
+## Configuration
+
+Plugins can have configuration files. These files are used to store data that the plugin needs to run. For example, a
+plugin that adds a new block to the game might have a configuration file that stores the block's ID. Configuration files
+are stored in the `plugins` folder in the server directory. Configuration files are written in YAML. See 
+[here](/paper/dev/plugin-configurations) for more information.
+
+## Scheduling Tasks
+
+Plugins can schedule tasks to run at a later time. This is useful for things like running code after a certain amount
+of time has passed. For example, a plugin might want to run code after 5 seconds. This can be done by scheduling a task
+to run after 100 ticks - one second is 20 ticks during normal operation. It is important to note that tasks might be 
+delayed if the server is lagging. For example, if the server is only running at 10 ticks per second, a task that is 
+scheduled to run after 100 ticks will take 10 seconds.
```

Comment:
this note here is good and I believe it should be added to the scheduler page too